docs/tools/flamegraphs.md

```diff
@@ -0,0 +1,167 @@
+# Generating Flamegraphs for a Running Node Service on Linux
+
+This guide assumes a running instance of Lodestar and will walk through how to generate a flamegraph for the process while running on Linux.
+
+## Modifying Lodestar
+
+Use the following two commands to install `perf` for generating the stack traces.  You may get a warning about needing to restart the VM due to kernel updates. This is nothing to be concerned with and if so, cancel out of the restart dialog.
+
+```bash
+sudo apt-get install linux-tools-common linux-tools-generic
+sudo apt-get install linux-tools-`uname -r`  # empirically this throws if run on the same line above
+```
+
+SSH into the Lodestar instance and modify `~/beacon/beacon_run.sh` to add a necessary flag.
+
+```sh
+admin@12.34.56.78: vim ~/beacon/beacon_run.sh # add --perf-basic-prof flag
+admin@12.34.56.78: cat ~/beacon/beacon_run.sh # should look something like the example below
+```
+
+### Example beacon_run.sh
+
+```sh
+#!/bin/bash
+
+# Load location of node to bin path
+source ~/.nvm/nvm.sh
+
+# Allows to edit node args, and lodestar args
+# To apply changes, restart the systemd service
+# ```
+# systemctl restart beacon
+# ```
+#
+# DON'T FORGET '\' CHARACTER WHEN EDITING FLAGS!!
+
+node \
+  --perf-basic-prof \
+  --max-old-space-size=4096 \
+  /usr/src/lodestar/packages/cli/bin/lodestar \
+  beacon \
+  --rcConfig /home/devops/beacon/rcconfig.yml
+```
+
+After updating `beacon_run.sh` file, restart the beacon service.
+
+```sh
+admin@12.34.56.78: sudo systemctl restart beacon.service
```

Comment:
Lodestar may be run with different service names, you should specify at the beginning what environment you are assuming for this guide

Reply:
Updated.